### ISSUE TITLE
The `find` method returns an asynchronous generator of dictionaries

### ISSUE THREAD

**Issue by returnnullptr** [Scille/umongo#294]
Hi there,

Since [Motor 2.2](https://motor.readthedocs.io/en/stable/changelog.html?highlight=next#motor-2-2):

> Added the AsyncIOMotorCursor method next() that advances the cursor one document at a time, similar to to the AsyncIOMotorChangeStream method next().

> Deprecated AsyncIOMotorCursor method next_object() and property fetch_next. Applications should use async for to iterate over cursors instead.

From this follows the issue:

```python
import asyncio

from motor.motor_asyncio import AsyncIOMotorClient
from umongo import Document, fields, MotorAsyncIOInstance


instance = MotorAsyncIOInstance()


@instance.register
class Example(Document):
    id = fields.StringField(attribute='_id')
    value = fields.StringField()


async def main(host: str):
    client = AsyncIOMotorClient(host)
    instance.init(client.get_default_database())

    await Example(id='1', value='1').commit()

    document = await Example.find_one({})
    assert isinstance(document, Example)

    for document in await Example.find({}).to_list(None):
        print('Synchronous iteration:', document)
        assert isinstance(document, Example)

    async for document in Example.find({}):
        print('Asynchronous iteration:', document)
        assert isinstance(document, Example)  # <-- This assertion failed

    client.close()


if __name__ == '__main__':
    asyncio.run(main('mongodb://localhost/test'))

```

Console output:

```
Synchronous iteration: <object Document __main__.Example({'id': '1', 'value': '1'})>
Asynchronous iteration: {'_id': '1', 'value': '1'}
```

Environment:

```console
(.venv) astile@strix-laptop:~/projects/umongo_bug$ uname -a
Linux strix-laptop 5.4.0-42-generic #46~18.04.1-Ubuntu SMP Fri Jul 10 07:21:24 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

(.venv) astile@strix-laptop:~/projects/umongo_bug$ python -V
Python 3.7.5

(.venv) astile@strix-laptop:~/projects/umongo_bug$ pip freeze
marshmallow==2.21.0
motor==2.2.0
pymongo==3.11.0
python-dateutil==2.8.1
six==1.15.0
umongo==2.2.0
```

Workaround: downgrade Motor to `motor==2.1.0`

Thanks!

**Comment by lafrech** [comment]
Thanks for reporting.

Is there something we should fix in `WrappedCursor`?

Do you think you could help with this?

umongo 3 is in beta state, so we can fix it for Motor 2.2 and drop compatibility with former versions.

**Comment by atti92** [comment]
I have the same problem

For umongo 2: should explicitly require motor<2.2.0) as a hotfix, in the extras section.

**Comment by lafrech** [comment]
Thank you guys for your help on this.

I created a 2.x branch. I'd rather not support umongo 2 eternally, but since umongo 3 is still beta and the transition might not be straightforward, I don't mind issuing at least bugfix releases. Like we do in marshmallow, I'll merge 2.x in master each time it is updated. I fixed compatibility with latest pytest in 2.x then merged it in master.

There is still an issue with txmongo tests. This is txmongo not being compatible with pymongo 3.11. This was addressed in txmongo in https://github.com/twisted/txmongo/commit/63dba8e85bca4677edb4be741a68a755f9349aea but there was no txmongo release since. I don't think this is a umongo issue. We could, and should, pin the dependencies versions in the tests. Until we do so, I don't think there is much we can do but wait.

In fact, pinning dependencies and using dependabot would be really nice to spot this kind of issues before they pile up.

Now, back to this specific issue. Thanks @atti92 for the PR. If the fix can be easily applied to umongo2 without a breaking change (without dropping older motor versions), then please send a PR against 2.x. Otherwise, we should require motor<2.2.0 in 2.x as you suggest (PR welcome too) and drop older motor versions in 3.x (master).

In both cases, please rebase #295 (against 2.x or master) to get the pylint compat fix, add a non-reg test (IIUC, this issue doesn't break any existing test), and please add yourself to AUTHORS file.

Thanks again.

**Comment by atti92** [comment]
This change should be backward compatible based on my testing. 
(Also added a little type checking for **async for** to a testcase.

**Comment by lafrech** [comment]
Fixed and released in 2.3.0.

Fixed in master.
